### PR TITLE
hubble: Add support for Cilium debug events

### PR DESCRIPTION
**Pull request by gandro**
This adds server-side support for debug events in Hubble. So far, they have only been available in `cilium monitor -D -v`.

 To be able to observe these events, one needs to enable debug events as documented here: https://docs.cilium.io/en/v1.9/operations/troubleshooting/#enabling-datapath-debug-messages 

``` console
$ grpcurl -plaintext -d '{"number":30,"whitelist":{"event_type":{"type": 2}}}' -unix /var/run/cilium/hubble.sock observer.Observer.GetFlows
{
  "debugEvent": {
    "type": "DBG_IP_ID_MAP_SUCCEED4",
    "source": {
      "ID": 2460,
      "identity": 4,
      "labels": [
        "reserved:health"
      ]
    },
    "hash": 705182630,
    "arg1": 3909094154,
    "arg2": 1,
    "arg3": 0,
    "message": "Successfully mapped addr=10.11.0.233 to identity=1",
    "cpu": 0
  },
  "nodeName": "k8s1",
  "time": "2021-01-13T14:30:11.896863575Z"
}

```

~Draft because blocked on #14432~

### CODE REVIEW

**Review by rolinh — APPROVED** [review]
LGTM!

File: daemon/cmd/hubble.go [hunk 1]
```diff
@@ -260,6 +260,15 @@ func (d *Daemon) GetEndpointInfo(ip net.IP) (endpoint v1.EndpointInfo, ok bool)
 	return ep, true
 }
 
+// GetEndpointInfo returns endpoint info for a given Cilium endpoint id. Used by Hubble.
+func (d *Daemon) GetEndpointInfoByID(id uint16) (endpoint v1.EndpointInfo, ok bool) {
+	ep := d.endpointManager.LookupCiliumID(id)
+	if ep == nil {
+		return nil, false
+	}
+	return ep, true
```
Comment (kaworu) [inline daemon/cmd/hubble.go:269]: nitpicking:
```suggestion
	return ep, (ep != nil)
```

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
restart-runtime

**Comment by gandro** [comment]
retest-runtime

**Review by kaworu — APPROVED** [review]

**Comment by gandro** [comment]
Runtime hit the issue fixed in #14601 - rebasing to ensure this does not occur again.

File: daemon/cmd/hubble.go (hunk 1 above)
Comment (gandro) [reply daemon/cmd/hubble.go:269]: I'll leave as is for consistency with the other getters in that file (which also use the `x == nil` if statement pattern).

**Comment by gandro** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

**Pull request merged**